README.md

```diff
@@ -1,61 +1,236 @@
+| -out rootCA.crt | Save the root certificate to the `rootCA.crt` file. |
+
+The [`openssl x509 -req`](https://docs.openssl.org/master/man1/openssl-x509/) command indicates to expect a self-signed PKCS#10 certificate request.
+
+| Option | Explanation |
+|--------|-------------|
+| -CA rootCA.crt | Specifies the "CA" certificate to be used for signing. When present, this behaves like a "micro CA" as follows: The subject name of the "CA" certificate is placed as issuer name in the new certificate, which is then signed using the key specified by the `-CAkey` option. |
+| -CAkey rootCA.key | Sets the CA private key to sign a certificate with. The private key must match the public key of the certificate specified by the `-CA` option.  |
+| -in kms-signing.csr | Read the certificate request from `kms-signing.csr` file. |
+| -out kms-signing.crt | Write output to `kms-signing.crt` file. |
+| -days 365 | Specifies that the newly-generated certificate expires in 365 days. |
+| -copy_extensions copyall | Copy all extensions, except that subject identifier and authority key identifier extensions. |
+
+## Create certificate chain
+
+Create certificate chain file PEM with certificate issued by CA and CA Root certificate. For example, with the self-signed certificate:
 
-```shell
-FLASK_KMS_KEY_ID={KMS_KEY} FLASK_CERT_CHAIN_PATH={CERT_CHAIN_PATH} flask run
 ```
-
-Send a JPEG to `http://localhost:5000/attach` to have it signed.
-
-```
-$ curl -X POST -T "<path to your jpg>" -o attached.jpg 'http://localhost:5000/attach'
+cat kms-signing.crt rootCA.crt > chain.pem
 ```
+
+## Run the application 
+
+1. Run the application by entering this command:
+    ```
+    FLASK_KMS_KEY_ID="$KMS_KEY_ID" FLASK_CERT_CHAIN_PATH="./chain.pem" flask run
+    ```
+    You'll see a response like this:
+    ```
+    Using KMS key: cdd59e61-b6fa-4d95-b71f-8d6ae3abc123
+    Using certificate chain: ./chain.pem
+    * Debug mode: off
+    WARNING: This is a development server. Do not use it in a production deployment. 
+    Use a production WSGI server instead.
+    * Running on http://127.0.0.1:5000
+    Press CTRL+C to quit
+    ```
+2. Upload and sign image: In another terminal window, use `curl` to upload an image file (the app works only with JPEGs) and have the app sign it by entering a command like this:
+    ```
+    curl -X POST -T "<PATH_TO_JPEG>" -o <SIGNED_FILE_NAME>.jpg 'http://localhost:5000/attach'
+    ```
+    For example:
+    ```
+    curl -X POST -T ~/Desktop/test.jpeg -o signed.jpeg 'http://localhost:5000/attach' 
+    ```
+    In this example, the image with signed Content Credentials is saved to `signed.jpeg`.
+
+If you encounter any issues running the `curl` command, try using `127.0.0.1` instead of `localhost`.
+
+Confirm that the app signed the output image:
+- If you've installed C2PA Tool, run `c2patool <SIGNED_FILE_NAME>.jpg`.
+- Uploading the image to https://contentcredentials.org/verify.
```

Comment:
May want to add a note than an image uploaded to the Verify site will appear as untrusted if it was signed with a self-signed cert. The important part is that the metadata appears correctly.

Reply:
Added a note.